src/components/button/Button.tsx

```diff
@@ -21,10 +21,14 @@ export interface ButtonProps {
     isDisabled?: boolean;
     /** whether the button is loading or not */
     isLoading: boolean;
+    /** toggles responsive behavior */
+    isResponsive?: boolean;
     /** whether the button is selected or not */
     isSelected?: boolean;
     /** onClick handler for the button */
     onClick?: Function;
+    /** component to show when breakpoint is reached */
+    responsiveContent?: React.ReactNode;
```

Comment:
Does it work to combine the component from this prop with `children`?

Reply:
i.e. in the parent of this component, it's setting `isResponsive` and `responsiveContent`. Does it make sense that when the values of `isResponsive` and `responsiveContent` is supposed to change, it will instead swap the components passed through `children`?

This way we won't need to update separate props.

Reply:
Thanks for reviewing! Not sure I understand correctly here, but if we were to combine `responsiveContent` with `children` we would probably need to use something like `useMediaQuery`, which might not be appropriate as I haven't seen it being used anywhere else in this repo. Or we could simply discard this change and let parents fully control what they want to show on different screen sizes by passing in dynamic `children`.
But for now, I've removed `responsiveContent` and reused the existing `icon`. Also made it backward compatible, so it wouldn't affect existing parents as long as `isResponsive` is false.